alea/include/alps/alea/internal/pooling.hpp

```diff
@@ -0,0 +1,102 @@
+#include <alps/alea/core.hpp>
+#include <alps/alea/util.hpp>
+
+#include <alps/alea/variance.hpp>
+#include <alps/alea/covariance.hpp>
+#include <alps/alea/autocorr.hpp>
+
+#include <alps/alea/internal/joined.hpp>
+
+namespace alps { namespace alea { namespace internal {
+
+template <typename T, typename Str>
+const typename eigen<typename traits<cov_result<T,Str>>::cov_type>::matrix &
+get_cov(const cov_result<T,Str> &result)
+{
+    return result.cov();
+}
+
+template <typename Result>
+typename eigen<typename traits<Result>::cov_type>::matrix
```

Comment:
> Why is the return type different from the previous overload (value vs const ref)?

For performance reasons. In the above case, the covariance is stored and can be directly returned, in the other it is constructed on-demand.  This is okay I think, since this is a very much internal method.